Dodge.js

```diff
@@ -1,178 +1,201 @@
 import { StatusBar } from 'expo-status-bar';
+import React, { Component, PureComponent } from 'react';
 import { StyleSheet, Text, View, Dimensions, Button, Image } from 'react-native';
-import { Knight, Circle, Brick, Grenade, MoveKnight, Shrapnel, counter} from "./Entities.js";
+import {
+  Knight, Circle, Brick, Grenade, MoveKnight,
+  Shrapnel, Water
+} from "./Entities.js";
 import { GameEngine } from "react-native-game-engine";
+import Matter from "matter-js";
 import AxisPad from 'react-native-axis-pad';
 ////this looks interesting for the attacker placing/throwing projectiles:
 //https://reactnative.dev/docs/panresponder 
 //https://medium.com/yapsody-engineering/hooks-and-function-components-in-react-native-d4e667c90cda
-var ents = {}
+let score = 0
+export default function DodgeBallGame({ navigation }) {
+  return (
+      <DodgeBall nav = {navigation}/>
+  )  
+}
 
-const generateUniqueId = require('generate-unique-id');
+class DodgeBall extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.ents = {}
+    this.counter = 0
+    this.nav = props.nav
+    const generateUniqueId = require('generate-unique-id');
+    this.MAXwidth = Dimensions.get('window').width;
+    this.MAXheight = Dimensions.get('window').height;
+    //this.deleteEntity = this.deleteEntity.bind(this)
+    this.addEntity(new Knight(this));
+    //this.addEntity(new Circle(this));
+    this.addEntity(new Brick(this));
+    this.addEntity(new Grenade(this));
+    this.addEntity(new Brick(this));
+    this.addEntity(new Circle(this));
+    this.addEntity(new Grenade(this));
+    this.addEntity(new Water(this))
+    setTimeout(() => {this.nav.navigate("Battle")}, 5000)
 
-const MAXwidth = Dimensions.get('window').width;
-const MAXheight = Dimensions.get('window').height;
 
+  }
 
-//adds an entity given its type, if it's a Grenade set a timer for explosion
-function AddEntity(entity){
-let id = generateUniqueId()
-ents[id] = new entity(id)
+  // this.isLetterInWord = this.isLetterInWord.bind(this);
 
-if(entity == Grenade){
-  setTimeout(() => { Explode(id) }, 2000)
-}
-}
 
-//deletes a single entity given that entities unique id
-function DeleteEntity(id){
-delete(ents[id])
-}
+  render() {
+    return (
+      <View>
```

Comment:
You probably don't need a view if has no styles and it contains a single component. It probably doesn't hurt either.